couchdb/src/test/java/org/hibernate/ogm/datastore/couchdb/test/embeddable/EmbeddableMappingTest.java

```diff
@@ -0,0 +1,361 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.couchdb.test.embeddable;
+
+import static org.hibernate.ogm.datastore.couchdb.utils.CouchDBTestHelper.assertDbObject;
+
+import java.util.Arrays;
+
+import org.hibernate.Transaction;
+import org.hibernate.ogm.OgmSession;
+import org.hibernate.ogm.backendtck.embeddable.Account;
+import org.hibernate.ogm.backendtck.embeddable.Address;
+import org.hibernate.ogm.backendtck.embeddable.AddressType;
+import org.hibernate.ogm.backendtck.queries.Ending;
+import org.hibernate.ogm.backendtck.queries.OptionalStoryBranch;
+import org.hibernate.ogm.backendtck.queries.StoryBranch;
+import org.hibernate.ogm.backendtck.queries.StoryGame;
+import org.hibernate.ogm.utils.OgmTestCase;
+import org.junit.Test;
+
+/**
+ * Tests for {@code @Embeddable} types and {@code @ElementCollection}s there-of.
+ *
+ * @author Emmanuel Bernard
+ * @author Gunnar Morling
+ * @author Marco Rizzi
+ */
+public class EmbeddableMappingTest extends OgmTestCase {
+
+	@Test
+	public void testEmbeddable() throws Exception {
+		OgmSession session = openSession();
+
+		Transaction transaction = session.beginTransaction();
+
+		// Given, When
+		Account account = new Account();
+		account.setLogin( "emmanuel" );
+		account.setPassword( "like I would tell ya" );
+		account.setHomeAddress( new Address() );
+		final Address address = account.getHomeAddress();
+		address.setCity( "Paris" );
+		address.setCountry( "France" );
+		address.setStreet1( "1 avenue des Champs Elysees" );
+		address.setZipCode( "75007" );
+		address.setType( new AddressType( "main" ) );
+		session.persist( account );
+		transaction.commit();
+
+		session.clear();
+
+		transaction = session.beginTransaction();
+
+		// Then
+		assertDbObject(
+				session.getSessionFactory(),
+				// collection
+				"Account",
+				// query
+				"Account:login_:emmanuel_",
+				// expected
+				"{ " +
+					"'login' : 'emmanuel', " +
+					"'homeAddress' : {" +
+						"'city' : 'Paris', " +
+						"'country' : 'France', " +
+						"'street1' : '1 avenue des Champs Elysees'," +
+						"'type' : {" +
+							"'name' : 'main'" +
+						"}" +
+					"}, " +
+					"'postal_code' : '75007', " +
```

Comment:
Could you add a TODO comment pointing to OGM-893? That'll avoid potential confusion as this is not really what one would expect.